rebasehelper/output_tools/text_output_tool.py

```diff
@@ -62,54 +62,82 @@ def print_changes_patch(cls):
     @classmethod
     def print_message_and_separator(cls, message="", separator='='):
         logger_report.info(message)
-        logger_report.info(separator * len(message))
+        logger_report.info(separator * (len(message)-1))
 
     @classmethod
-    def print_patches(cls, patches, summary):
+    def print_patches(cls, patches):
+        cls.print_message_and_separator("\nDownstream Patches")
+        logger_report.info("Legend:")
+        logger_report.info("[-] = already applied, patch removed")
+        logger_report.info("[*] = merged, patch modified")
+        logger_report.info("[!] = conflicting or inapplicable, patch skipped")
+        logger_report.info("[ ] = patch untouched")
         if not patches:
             logger_report.info("Patches were neither modified nor deleted.")
             return
-        logger_report.info(summary)
-        max_name = 0
-        for value in six.itervalues(patches):
-            if value:
-                new_max = max([len(os.path.basename(x)) for x in value])
-                if new_max > max_name:
-                    max_name = new_max
-        max_key = max([len(x) for x in six.iterkeys(patches)])
-        for key, value in six.iteritems(patches):
-            if value:
-                for patch in value:
-                    logger_report.info('Patch %s [%s]', os.path.basename(patch).ljust(max_name), key.ljust(max_key))
+
+        for patch_type, patch_list in six.iteritems(patches):
+            if patch_list:
+                for patch in patch_list:
+                    if patch_type == 'deleted':
+                        logger_report.info(' * {0:40} [-]'.format(os.path.basename(patch)))
+                    elif patch_type == 'modified':
+                        logger_report.info(' * {0:40} [*]'.format(os.path.basename(patch)))
+                    elif patch_type == 'inapplicable':
+                        logger_report.info(' * {0:40} [!]'.format(os.path.basename(patch)))
+                    else:
+                        logger_report.info(' * {0:40} [ ]'.format(os.path.basename(patch)))
```

Comment:
You could use mapping dict, something like:

```python
symbols = dict(deleted='-', modified='*', inapplicable='!')
```

and then reference it like this:

```python
symbols.get(patch_type, ' ')
```

Reply:
Awesome, thanks!